rust/src/db.rs

```diff
@@ -24,6 +30,15 @@ pub type SqliteConnection = PoolConnection<Sqlite>;
 /// Wallet has to be managed by Rust as generics are not support by frb
 static DB: Storage<Mutex<SqlitePool>> = Storage::new();
 
+fn into_array(vec: Vec<u8>) -> [u8; 32] {
+    let slice = vec.as_slice();
+    let array: [u8; 32] = match slice.try_into() {
+        Ok(arr) => arr,
+        Err(_) => panic!("Expected a Vec of length {} but it was {}", 32, vec.len()),
+    };
+    array
```

Comment:
🙃 StackOverflow offers a simpler version: https://stackoverflow.com/a/29570662

```rust
v.try_into()
        .unwrap_or_else(|v| panic!("Expected a Vec of length 32 but it was {}", v.len()))
```

Does that not work?

Reply:
indeed - great suggestion!

also: I thought anyone would comment on the use of `map()` for value inside the option, I only recently found that out and made things way more readable :D